src/client/components/OIbForm/OIbCheckBox.jsx

```diff
@@ -13,6 +16,19 @@ const OIbCheckBox = ({ label, value, name, onChange, defaultValue, switchButton
     const { checked } = target
     onChange(name, checked, null)
   }
+  if (dependency) {
+    // check if dependency exists, and compare the value
```

Comment:
is this test necessary as the hook UseEffect is controlled by [dependecyValue]?

Reply:
The "if (dependency)" test is needed because we only need to check the dependencyValue if we have a dependency (as path) defined, and the hook is controlled by  dependencyValue which exists based on the path.